### ISSUE TITLE
Support multiple dimensions in cross, pinv and solve

### ISSUE THREAD

**Issue by sob3kx** [QuantStack/xtensor-blas#124]
**Cross**
At the moment, cross seems to be limited to 1D vectors of size 2 or 3. Numpy doesn't have the constraint for dimensionality and is capable of computing cross for higher dimensionality arrays if the last dimension has length of 2 or 3.

xt::xarray<double> firstArray = xt::ones<double>({ 3, 4, 2 });
xt::xarray<double> secondArray = xt::ones<double>({ 3, 4, 2 });

auto result = xt::linalg::cross(firstArray, secondArray);
std::runtime_error("a or b did not have appropriate size 2 or 3.");

a = numpy.ones([3, 4, 2])
b = numpy.ones([3, 4, 2])
numpy.cross(a, b)
array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]])
numpy.cross(a, b).shape
(3, 4)

**Pinv**

Pinv seems to support only 1D or 2D arrays. NumPy does support higher dimensionality.

xt::xarray<double> a = xt::ones<double>({ 1, 2, 3, 4});

auto result = xt::linalg::pinv(a);

XTENSOR_ASSERT_MSG(N == e.derived_cast().dimension(), "Cannot change dimension of xtensor.");

numpy.linalg.pinv(numpy.ones([1, 2, 3, 4]))
array([[[[0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333]],

        [[0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333],
         [0.08333333, 0.08333333, 0.08333333]]]])

a = numpy.array([[[1, 2, 3]]])
a.shape
(1, 1, 3)
numpy.linalg.pinv(a)
array([[[0.07142857],
        [0.14285714],
        [0.21428571]]])

**Solve**
Solve seems to be limited to <= 2D by XTENSOR_ASSERT. 
auto a = xt::xarray<double>({ {0.72999173, 0.94937934, 0.6160863},
                            {0.67719638, 0.11662088, 0.86027322},
                            {0.22839025, 0.14736096, 0.36511296} });
auto b = xt::xarray<double>({ {{0.42682534, 0.18769713},
                            {0.09106305, 0.93373785},
                            {0.3472225 , 0.05267208}},

                            {{0.09015187, 0.94741843},
                            {0.85322852, 0.25576913},
                            {0.47621316, 0.5724512}} });
auto result = xt::linalg::solve(a, b);
XTENSOR_ASSERT failed.

a = numpy.random.rand(9).reshape(3,3)
b = numpy.random.rand(12).reshape(2, 3, 2)
numpy.linalg.solve(a, b)
array([[[-3.08713549,  3.33567358],
        [ 1.29121029, -1.49942674],
        [ 2.36096848, -1.33713771]],

       [[-1.99462003, -4.35353126],
        [-0.03715958,  2.11458264],
        [ 2.56698863,  3.43769886]]])

Having ability to compute this functions with higher dimensionality arrays would help anyone doing more demanding calculations.

**Comment by wolfv** [comment]
yes, we plan to support this functionality by taking advantage of views. we want (and already have prototyped) a view-iterator. 

The iterator will work something like this:

auto [it_start, it_end] = xt::blockiter_from_end<2>(my_array);

will yield 2D `xview`s on the underlying array. Those can then easily be passed to the functions mentioned above.
